SPECS/golang/golang.spec

```diff
@@ -1,6 +1,6 @@
 %global goroot          %{_libdir}/golang
 %global gopath          %{_datadir}/gocode
-%global ms_go_filename  go1.22.6-20240806.4.src.tar.gz
+%global ms_go_filename  go1.23.0-20240814.2.src.tar.gz
```

Comment:
N-Ack
There are a few packages that fail and we may want to investigate:
[PR-10161](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=624784&view=logs&j=eb4d3d72-7ec0-56f0-d57c-e2656014f3b8&t=bb9783d9-f9cf-5bda-3956-217e117e7ec9&l=4229)